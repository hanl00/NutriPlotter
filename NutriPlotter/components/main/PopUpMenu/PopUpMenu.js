import React, { Component } from 'react'
import { Alert, Button, StyleSheet, View, TouchableOpacity, Text } from 'react-native'
import { Feather } from '@expo/vector-icons';
import {
  Menu,
  MenuProvider,
  MenuOptions,
  MenuOption,
  MenuTrigger,
} from 'react-native-popup-menu';

import styles from './styles';
const PopUpMenu = () => (
   <MenuProvider style={styles.menuhead}>
     <Menu onSelect={value => alert(`Selected number: ${value}`)}>
     <MenuTrigger>
      <Feather
         name = 'menu'
         size = {50}
         color = {'black'}
         />
     </MenuTrigger>
       <MenuOptions>

         <MenuOption value ={666}>
            <Text style={styles.TextStyle}>Start Over</Text>
         </MenuOption>

         <MenuOption value ={2}>
            <Text style={styles.TextStyle}>Exit</Text>
         </MenuOption>

       </MenuOptions>
     </Menu>
   </MenuProvider>
 );



export default PopUpMenu;
