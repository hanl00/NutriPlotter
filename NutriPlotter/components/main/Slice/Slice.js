import React, {Component} from 'react';
import {Svg} from 'expo';
import * as shape from 'd3-shape';
const d3 = {shape};

export default class Slice extends Component {
    constructor(props) {
        super(props);
        this.state = {};
        this.arcGenerator = d3.shape.arc()
            .outerRadius(100)
            .padAngle(0)
            .innerRadius(0);
    }

    createPieArc = (index,startAngle, endAngle) => {

        var arc = d3.shape.arc()
        .innerRadius(0)
        .outerRadius(100)
        .startAngle(startAngle)
        .endAngle(endAngle);



        return arc();
    };


    render() {

        const {
            endAngle,
            color,
            index,
            startAngle,
        } = this.props;

        return (

            <Svg.Path
                onPress={()=>this.props.pressHandler(index)}
                d={this.createPieArc(index,startAngle, endAngle)}
                fill={color}
                cx={12} cy={12}
            />
        )

    }
}
