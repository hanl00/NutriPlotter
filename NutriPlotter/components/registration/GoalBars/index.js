import CarbohydrateBar from './CarbohydrateBar';
import FatBar from './FatBar';
import ProteinBar from './ProteinBar';
import styles from './styles';

export {CarbohydrateBar, FatBar, ProteinBar, styles};
