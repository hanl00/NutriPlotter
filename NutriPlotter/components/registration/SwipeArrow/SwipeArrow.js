import React from 'react';
import PropTypes from 'prop-types';

import {View, Image, Text} from 'react-native';

import styles from './styles';

//stateless function!
const SwipeArrow = ({imageSource}) => (
    <View
      style={styles.container}
    >
    <Text style={styles.swipeText}>Swipe to next page</Text>
    <Image source={imageSource}/>

    </View>
);

SwipeArrow.propTypes = {
  imageSource: Image.propTypes.source,
}

export default SwipeArrow;
