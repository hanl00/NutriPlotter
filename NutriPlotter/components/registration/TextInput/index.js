import RegistrationTextInput from './RegistrationTextInput';
import styles from './styles';

export {RegistrationTextInput, styles};
