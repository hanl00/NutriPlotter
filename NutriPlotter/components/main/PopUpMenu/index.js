import PopUpMenu from './PopUpMenu';
import styles from './styles';

export {PopUpMenu, styles};
