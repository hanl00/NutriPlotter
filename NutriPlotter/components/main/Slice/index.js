import Slice from './Slice';
import styles from './styles';

export {Slice, styles};
