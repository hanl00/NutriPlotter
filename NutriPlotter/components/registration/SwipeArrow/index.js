import SwipeArrow from './SwipeArrow';
import styles from './styles';

export {SwipeArrow, styles};
