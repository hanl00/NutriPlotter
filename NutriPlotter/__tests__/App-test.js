import {
  NativeModules,
  LayoutAnimation,
  Text,
  View,
  Button,
  Image,
  TouchableOpacity,
  Animated,
  PanResponder,
  Dimensions,
  Platform,
  Alert,
  BackHandler,
  NativeEventEmitter
} from 'react-native';
import * as React from 'react';
//import App from '../App.js';
import * as renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import { FoodList } from '../components/main/FoodList'
import { FoodItem } from '../components/main/FoodItem'
import { Slice } from '../components/main/Slice'
import { GoalBars } from '../components/registration/GoalBars'
import { PopUpMenu } from '../components/main/PopUpMenu'
import { SwipeArrow } from '../components/registration/SwipeArrow'
import { RegistrationTextInput } from '../components/registration/TextInput'
import { RegistrationTextInputWithSwitch } from '../components/registration/TextInputWithSwitch'
import { Svg } from 'expo'

it('Food list test against snapshot', () => {
const tree = renderer.create(<FoodList></FoodList>).toJSON();
expect(tree).toMatchSnapshot();
});

it('FoodItem test against snapshot (using chicken)', () => {
const tree = renderer.create(<FoodItem name={'Chicken'} src={require('../assets/images/chicken.png')}/>).toJSON();
expect(tree).toMatchSnapshot();
});

it('SwipeArrow test against snapshot', () => {
const tree = renderer.create(<SwipeArrow imageSource={require('../screens/registration/src/arrows.png')}/>).toJSON();
expect(tree).toMatchSnapshot();
});


it('RegistrationTextInput test against snapshot', () => {
const tree = renderer.create(<RegistrationTextInput
  textPH = {"name"}
  updateState = {this.updateNameState}
  canUpdateState = {false}
/>).toJSON();
expect(tree).toMatchSnapshot();
});

it('RegistrationTextInputWithSwitch test against snapshot', () => {
const tree = renderer.create(<RegistrationTextInputWithSwitch
  textPH = {"Age"}
  handleOptionA = {this.handleOptionA}
  handleOptionB = {this.handleOptionB}
  styleSwitcher = {null}
  updateState = {this.updateAgeState}
  canUpdateState = {false}
/>).toJSON();
expect(tree).toMatchSnapshot();
});
