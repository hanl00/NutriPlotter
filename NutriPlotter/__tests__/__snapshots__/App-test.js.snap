// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Food list test against snapshot 1`] = `null`;

exports[`FoodItem test against snapshot (using chicken) 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    aspectRatio={1}
    borderRadius={150}
  >
    <Image
      resizeMode="contain"
      source={1}
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Chicken
    </Text>
  </View>
</View>
`;

exports[`RegistrationTextInput test against snapshot 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="name"
    placeholderTextColor="black"
    textAlign="center"
    underlineColorAndroid="rgba(0,0,0,0)"
  />
</View>
`;

exports[`RegistrationTextInputWithSwitch test against snapshot 1`] = `
<View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {},
      ]
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 50,
            "height": 40,
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "#C9C9C9",
              "borderRadius": 50,
              "borderWidth": 0,
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#FFFFFF",
                      "fontSize": 14,
                      "fontWeight": "normal",
                      "textAlign": "center",
                    },
                    Object {},
                  ]
                }
              >
                Male
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "opacity": 1,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#000000",
                      "fontSize": 14,
                      "fontWeight": "normal",
                      "textAlign": "center",
                    },
                    Object {},
                  ]
                }
              >
                Female
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <TextInput
    allowFontScaling={true}
    keyboardType="numeric"
    onChangeText={[Function]}
    placeholder="Age"
    placeholderTextColor="black"
    textAlign="center"
    underlineColorAndroid="rgba(0,0,0,0)"
  />
</View>
`;

exports[`SwipeArrow test against snapshot 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Swipe to next page
  </Text>
  <Image
    source={1}
  />
</View>
`;
